### PR TITLE
improve test_generate_snap test case.

### LINKED ISSUE #3129: test failpoints_cases::test_snap::test_generate_snapshot failed
```
test failpoints_cases::test_snap::test_generate_snapshot ... thread 'failpoints_cases::test_snap::test_generate_snapshot' panicked at 'the directory "/tmp/test_cluster.lAernMFQcZBN" should be empty', tests/failpoints_cases/test_snap.rs:242:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:207
   3: std::panicking::default_hook
             at libstd/panicking.rs:223
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:402
   5: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:349
   6: failpoints::failpoints_cases::test_snap::must_empty_dir
             at tests/failpoints_cases/test_snap.rs:242
   7: failpoints::failpoints_cases::test_snap::test_generate_snapshot
             at tests/failpoints_cases/test_snap.rs:220
   8: failpoints::__test::TESTS::{{closure}}
             at tests/failpoints_cases/test_snap.rs:194
   9: core::ops::function::FnOnce::call_once
             at /checkout/src/libcore/ops/function.rs:223
  10: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1453
             at /checkout/src/libcore/ops/function.rs:223
             at /checkout/src/liballoc/boxed.rs:635
  11: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
```

### PR DESCRIPTION
**Pull request by hicqu**
The test case fails frequently. However maybe it's not a bug, just because the wait time is not long enough.
This PR increase max wait time from 2s to 5s, so that it can tolerate more disk latency.
And panic message is improvced.

Fix #3129 .

### CODE REVIEW

File: tests/failpoints_cases/test_snap.rs [hunk 1]
```diff
@@ -231,13 +231,17 @@ fn test_generate_snapshot() {
...
-            continue;
+    for _ in 0..500 {
+        thread::sleep(Duration::from_millis(10));
+        if fs::read_dir(&path).unwrap().count() == 0 {
+            return;
         }
-        return;
     }
-    panic!("the directory {:?} should be empty", path);
+    let entries = fs::read_dir(&path)
```
Comment (siddontang) [inline tests/failpoints_cases/test_snap.rs]: maybe here the path is empty?

File: tests/failpoints_cases/test_snap.rs [hunk 2]
```diff
@@ -231,13 +231,19 @@ fn test_generate_snapshot() {
...
-            continue;
+    for i in 0..500 {
+        thread::sleep(Duration::from_millis(10));
+        let entries = fs::read_dir(&path)
+            .and_then(|dir| dir.collect::<io::Result<Vec<_>>>())
+            .unwrap();
+        if entries.is_empty() {
+            return;
+        }
+        if i == 499 {
```
Comment (BusyJay) [inline tests/failpoints_cases/test_snap.rs]: This is weird. Why not put panic outside the loop?

**Comment by BusyJay** [comment]
Any updates?

File: tests/failpoints_cases/test_snap.rs (hunk 2 above)
Comment (siddontang) [reply tests/failpoints_cases/test_snap.rs]: +1

**Comment by hicqu** [comment]
ping @BusyJay @overvenus .

File: tests/failpoints_cases/test_snap.rs [hunk 3]
```diff
@@ -193,13 +193,20 @@ fn test_generate_snapshot() {
...
+    for _ in 0..500 {
+        thread::sleep(Duration::from_millis(10));
+        if fs::read_dir(&path).unwrap().count() == 0 {
+            return;
         }
-        return;
     }
-    panic!("the directory {:?} should be empty", path);
+
+    let entries = fs::read_dir(&path)
```
Comment (BusyJay) [inline tests/failpoints_cases/test_snap.rs]: Why not use count here?

Comment (BusyJay) [reply tests/failpoints_cases/test_snap.rs]: OK, get it.

**Pull request merged**